spyder_kernels/utils/nsview.py

```diff
@@ -459,7 +465,7 @@ def value_to_display(value, minmax=False, level=0):
 
     # Truncate display at 70 chars to avoid freezing Spyder
     # because of large displays
-    if len(display) > 70:
+    if not isinstance(display, NUMERIC_TYPES) and len(display) > 70:
```

Comment:
If you agree with my suggestion above, this needs to include `numeric_numpy_types` too.